src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -239,166 +239,178 @@ protected void install(InstallConfig installConfig) {
-                        stopSyncingShadows(true);
-                        startSyncingShadows(StartSyncInfo.builder().reInitializeSyncInfo(true).startSyncStrategy(true)
-                                .updateCloudSubscriptions(true).build());
-                    }
-                } catch (InvalidConfigurationException e) {
-                    serviceErrored(e);
-                }
-            });
+    Strategy replaceStrategyIfNecessary(Strategy currentStrategy, Strategy strategy) {
+        if (!currentStrategy.equals(strategy)) {
+            currentStrategy = strategy;
+            stopSyncingShadows(false);
+            syncHandler.setSyncStrategy(strategy);
+            startSyncingShadows(StartSyncInfo.builder().startSyncStrategy(true).build());
         }
+        return currentStrategy;
+    }
 
-        if (installConfig.configureRateLimitsConfig) {
-            Topics rateLimitTopics = config.lookupTopics(CONFIGURATION_CONFIG_KEY, CONFIGURATION_RATE_LIMITS_TOPIC);
-            rateLimitTopics.subscribe((what, newv) -> {
-                Map<String, Object> rateLimitsPojo = rateLimitTopics.toPOJO();
-                try {
-                    if (rateLimitsPojo.containsKey(CONFIGURATION_MAX_OUTBOUND_UPDATES_PS_TOPIC)) {
-                        int maxOutboundSyncUpdatesPerSecond = Coerce.toInt(rateLimitsPojo
-                                .get(CONFIGURATION_MAX_OUTBOUND_UPDATES_PS_TOPIC));
-                        Validator.validateOutboundSyncUpdatesPerSecond(maxOutboundSyncUpdatesPerSecond);
-                        iotDataPlaneClientWrapper.setRate(maxOutboundSyncUpdatesPerSecond);
-                    }
-
-                    if (rateLimitsPojo.containsKey(CONFIGURATION_MAX_TOTAL_LOCAL_REQUESTS_RATE)) {
-                        int maxTotalLocalRequestRate = Coerce.toInt(rateLimitsPojo
-                                .get(CONFIGURATION_MAX_TOTAL_LOCAL_REQUESTS_RATE));
-                        Validator.validateTotalLocalRequestRate(maxTotalLocalRequestRate);
-                        inboundRateLimiter.setTotalRate(maxTotalLocalRequestRate);
-                    }
-
-                    if (rateLimitsPojo.containsKey(CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC)) {
-                        int maxLocalShadowUpdatesPerThingPerSecond = Coerce.toInt(rateLimitsPojo
-                                .get(CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC));
-                        Validator.validateLocalShadowRequestsPerThingPerSecond(maxLocalShadowUpdatesPerThingPerSecond);
-                        inboundRateLimiter.setRate(maxLocalShadowUpdatesPerThingPerSecond);
-                    }
-                } catch (InvalidConfigurationException e) {
-                    serviceErrored(e);
-                }
-            });
+    private void configureSynchronization(Node newv) {
+        if (newv != null && !newv.childOf(CONFIGURATION_SYNCHRONIZATION_TOPIC)) {
+            return;
+        }
```

Comment:
You don't use `newv` in here for anything - why pass it in? This check could happen outside of `configureSynchronization` 

Passing it in makes it seem like this node is going to be used for getting the configuration

Reply:
That's how it was in the previous version of the PR where the check happened outside of the configure methods. Modified like this to address your comment about the config validation being a bit defensive so the configure methods check based on newv.

Reply:
got it - I reread the previous thread and can see why you made that change. I was thinking about a larger change where the synchandler and the rest reads their own config and then getting rid of that installConfig object